lib/trino-hdfs/src/main/java/io/trino/hdfs/HdfsEnvironment.java

```diff
@@ -68,6 +70,15 @@ public HdfsEnvironment(
         this.newDirectoryPermissions = config.getNewDirectoryFsPermissions();
     }
 
+    @PreDestroy
+    public void shutdown()
+            throws IOException
+    {
+        FileSystemFinalizerService.getInstance().shutdown();
```

Comment:
no need to create instance just in order to shut it down.

Reply:
There would never not be an instance created in normal usage.

Reply:
> there will always be an existing instance.

What if only HiveMetadata has been called, without any file-system related operations? (eg on coordinator)
it seems wasteful to start a thread during shutdown